src/osdc/Objecter.h

```diff
@@ -2328,6 +2340,13 @@ class Objecter : public md_config_obs_t, public Dispatcher {
     return i;
   }
 
+  /**
+   * get a random osd not listed in blacklist
+   * @note the primary osd will be returned even it is listed in the blacklist
+   *       if the target pool is a erasure pool, the primary osd is returned.
```

Comment:
Confusing because you mention primary osd twice.  Is this sufficient?

the primary osd will be returned even it is listed in the blacklist
if the target pool is a erasure pool